app/src/ui/delete-branch/delete-branch-dialog.tsx

```diff
@@ -96,6 +103,8 @@ export class DeleteBranch extends React.Component<
 
   private deleteBranch = async () => {
     const { dispatcher, repository, branch, onDeleted } = this.props
+
+    this.setState({ isDeleting: true })
```

Comment:
There is no need to reset the state back to false since the component exits after deleting